docs/usage/self-hosted-configuration.md

```diff
@@ -417,9 +417,17 @@ Read the documentation for [git commit --no-verify](https://git-scm.com/docs/git
 To learn more about Git hooks, read the [Pro Git 2 book, section on Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks).
 
 ## gitPrivateKey
-
-This should be an armored private key, so the type you get from running `gpg --export-secret-keys --armor 92066A17F0D1707B4E96863955FEF5171C45FAE5 > private.key`.
-Replace the newlines with `\n` before adding the resulting single-line value to your bot's config.
+This property is used for [GPG/PGP signing](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). 
+To generate a GPG key, run the following commands in the terminal:
+`gpg --gen-key`
+For the following prompts, enter the following:
+Real Name: `Renovate Bot`
+Email Address: `bot@renovateapp.com`
```

Comment:
I'm not sure this is actually correct, is it? It should match the git committer, not the "official" bot email address? Also the public key needs to be uploaded somewhere publicly such as GitHub so that it can be verified.

Reply:
The `privateKey` docs for self-hosted users [^docs] also mentions GPG to create a private key.

The `privateKey` docs and this PR have some differences:

- The `privateKey` docs uses different commands:
  - Our docs: `gpg --full-generate-key`, this PR: `gpg --gen-key`
  - Our docs:  `gpg --armor --export-secret-keys YOUR_NEW_KEY_ID > renovate-private-key.asc` , this PR: `gpg --export-secret-keys --armor [your public key] | awk -v ORS='\\n' '1' | pbcopy`
- Our docs use `RENOVATE_PRIVATE_KEY` as variable name, this PR: `RENOVATE_GIT_PRIVATE_KEY`

I'll let you review these differences, I don't know enough about signing to help with this. 😉 

[^docs]: https://docs.renovatebot.com/self-hosted-configuration/#privatekey

Reply:
Those are different options

Reply:
I wrote this for a self hosted on Gitlab. This configuration worked for me. I didn't know that there could be other scenarios. Can someone advise what it should say?

Reply:
I don't know enough about `gpg` or `pgp` to help out here. 😄 I'll let one of the maintainers help you. 😉

Reply:
@rarkins, does this email address also have to match what's in the renovate's [config](https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/.gitlab-ci.yml#L4)?  That's what I why I have it as `bot@renovateapp.com`

Reply:
I think it needs to match the git _committer_, which is not the same as `gitAuthor` in Renovate config.

Reply:
git committer and `gitAuthor` really seem like the same thing. When I commit something, I'm the author. 

For brevity and to avoid confusion during setup, maybe it would be beneficial to keep them the same?

Reply:
Your first sentence is incorrect, please google it